pkg/common/message_sink.go

```diff
@@ -0,0 +1,61 @@
+package common
+
+import (
+	"errors"
+	"sync/atomic"
+)
+
+// MessageSink is a helper struct that allows to send messages to a message sink.
+// The MessageSink abstracts the message sink which has a certain sender, so that
+// the sender does not have to be specified every time a message is sent.
+// At the same it guarantees that the caller can't alter the `sender`, which means that
+// the sender can't impersonate another sender (and we guarantee this on a compile-time).
+type MessageSink[SenderType comparable, MessageType any] struct {
+	// The sender of the messages. This is useful for multiple-producer-single-consumer scenarios.
+	sender SenderType
+	// The message sink to which the messages are sent.
+	messageSink chan<- Message[SenderType, MessageType]
+	// Atomic variable that indicates whether the message sink is sealed.
+	// This is used to prevent sending messages to a sealed message sink.
+	// The variable is atomic because it may be accessed from multiple goroutines.
```

Comment:
For clarity's sake, could we define what 'sealed' means here?

Reply:
Good point, will add a comment about that! Perhaps I need to rewrite the comment for this variable altogether.